### ISSUE TITLE
Password Kodi user

### ISSUE THREAD

**Issue by svriend** [Saiyato/volumio-kodi-plugin#34]
Hi Saiyato,

First of all, great plug-in, especially for coding/linux noobs like me.

I have a question, I've been trying to add a keymap and other settings from another Kodi installation but I don't have permission (via SSH or SCP) as the volumio user but I can't seem to find the kodi user password. It's probably somewhere in these files but I I've been digging through them with my limited knowledge and seem to be stuck. Hopefully you (or someone else) can nudge me in the right direction.

Kind regards,
Stefan

**Comment by Saiyato** [comment]
Hi svriend,

The user is added without setting a password, see this part of the script:

```
		#adduser kodi
		useradd --create-home kodi
		usermod -aG audio,video,input,dialout,plugdev,tty kodi
		usermod -aG kodi volumio
```

You could set a passwd for it, but I never needed it to edit the guisettings.xml, not sure if that counts for keymap as well. What platform are you running on?

**Comment by svriend** [comment]
Hi Saiyato,

Thanks for taking time.

I was accessing the Pi/Volumio with WinSCP (running on Win 10) loged in as " volumio" and tried to copy the files over. But then I get the following error message.

![Annotation 2020-04-30 182718](https://user-images.githubusercontent.com/34350083/80735011-59c45c80-8b10-11ea-9999-db5bb63c7635.png)

Kind regards,
Stefan Vriend

**Comment by Saiyato** [comment]
Hi Stefan,

I think this should solve your problem, just execute in the CLI
`sudo chmod -R 775 /data/configuration/miscellanea/Kodi/Configuration`

See explanation here: https://en.wikipedia.org/wiki/Chmod

TL;DR
It allows members of the kodi group (volumio is a member) to _also_ write to the directory. The first number is for the user (which already has a full permission set), the second is for the group and the third is unchanged (i.e. 5 -> read+execute).

Oh and please let me know if this helps ;) So I can patch the scripts

**Comment by svriend** [comment]
Hi Saiyato,

Thanks for your fast reply and suggestion.

Earlier I was trying to change/add things in a different directory /home/kodi/.kodi/userdata (I don't know if this was the wrong folder).

I tried the command for both folders (using PuTTY logged in as volumio). At first it couldn't find the folder but I checked in WinSCP and that was due to the capital K of Kodi (should be kodi). I was then able to run the command but it sadly still gives the same Permission denied error (even after reboot, for both directories).

Do you have an idea of what we might try next?

Kind regards,
Stefan

**Comment by Saiyato** [comment]
Hi Stefan,

The error should be different; something like "upload of file x was successful, but error occurred while setting the permission and/or timestamp". This means you either need to change the file owner, or just live with the error (how often will you upload the file?). 

The file is written successfully, but the second part (setting permissions/timestamp) fails. Also read here: https://winscp.net/eng/docs/message_preserve_time_perm

Can you confirm this? :)

**Comment by svriend** [comment]
Hi,
it's been a while since opening this issue but I think I didn't get it to work and switched to a dual boot with OSMC because I was also running a newer version of kodi on my laptop.
Sorry for never answering your question and thanks for the help.

Kind regards
